src/Option.ts

```diff
@@ -712,71 +530,62 @@ export const isNone = <A>(fa: Option<A>): fa is None<A> => {
  * const isA = getRefinement<C, A>(c => (c.type === 'B' ? some(c) : none)) // static error: Type '"B"' is not assignable to type '"A"'
  * ```
  *
- * @since 1.7.0
+ * @since 2.0.0
  */
 export const getRefinement = <A, B extends A>(getOption: (a: A) => Option<B>): Refinement<A, B> => {
-  return (a: A): a is B => getOption(a).isSome()
+  return (a: A): a is B => isSome(getOption(a))
 }
 
-const compact = <A>(fa: Option<Option<A>>): Option<A> => fa.chain(identity)
+const compact = <A>(fa: Option<Option<A>>): Option<A> => chain(fa, identity)
+
+const defaultSeparate = { left: none, right: none }
+
 const separate = <RL, RR>(fa: Option<Either<RL, RR>>): Separated<Option<RL>, Option<RR>> => {
-  if (fa.isNone()) {
-    return {
-      left: none,
-      right: none
-    }
-  }
-  const e = fa.value
-  if (e.isLeft()) {
-    return {
-      left: some(e.value),
-      right: none
-    }
-  }
-  return {
-    left: none,
-    right: some(e.value)
-  }
+  return getOrElse(
+    map(fa, e => ({
+      left: getLeft(e),
+      right: getRight(e)
+    })),
+    defaultSeparate
+  )
 }
 
-const filter = <A>(fa: Option<A>, p: Predicate<A>): Option<A> => fa.filter(p)
+const filter = <A>(fa: Option<A>, p: Predicate<A>): Option<A> => (isNone(fa) ? fa : p(fa.value) ? fa : none)
```

Comment:
@gcanti I think we still need also the equivalent of the old `.filter` method, accepting a `Refinement<A, B>`. This is some code that doesn't compile anymore:
```ts
import { option, some } from 'fp-ts/lib/Option';

declare const a: string | number;
declare function isString(x: any): x is string;

const b: string = option.filter(some(a), isString);
```

Reply:
@giogonzo I think that we can handle refinements (in `filter` and `partition`) once for all by adding proper overloads to the `Filterable` type class. This is beneficial for v1 too, I'll send a PR for v.1.17.2